### DISCUSSION TITLE
GPS lost when screen is turned off with LineageOS 17 (Android 10)

### DISCUSSION THREAD

**Discussion by schubeda** [osmandapp/OsmAnd#11596] [General]
During navigation, when I turn off the screen and later turn it back on, I need to wait some time until location is found again.

Interestingly, it works if another navigation app (HERE WeGo) is doing navigation in parallel.

Energy optimizations for OsmAnd are switched off in the settings.

Android: LineageOS 17 (MicroG)
Device: Samsung Galaxy A3 2016
OsmAnd: 3.9.10 (f-droid)

Edit: No Google Play Services

**Comment by schubeda** [comment]
Really no one else with this problem?

It seems like a bug, since the same functionality works with another app.

Makes it unusable for longer navigation if I need to keep the screen on.

**Reply by vshcherb** [reply]
Unfortunately, we will need more feedback of course but you can try different (and old builds) that might help at least to have a working version.

**Comment by franco999** [comment]
The same thing has been happening to me for several versions. When I'm cycling and I'm recording a track, I keep my cell phone in my pocket and if after a few kilometers I want to see the route I did, I notice that there are sections where I didn't record anything.

**Reply by Lee-Carre** [reply]
I suggest using a dedicated track-recording app. You can then import the GPX into OsmAnd or whatever else you'd like.

Many such logging apps enable you to view the path of the currently recording track, against a background map.

**Reply by schubeda** [reply]
@franco999 Check the [troubleshooting](https://docs.osmand.net/en/main@latest/osmand/troubleshooting/track-recording-issues) for track recording. Maybe there is something that works for you.

@Lee-Carre There is a track recording plugin for Osmand, which is intended to work. Using another app is practical, but doesn't solve any problems with Osmand. If Osmand is not working properly, I think it should be our aim to fix these problems.

**Reply by franco999** [reply]
I recently switched to a ROM with **Android 11, OsmAnd +4.1** and tested again. Keeping OsmAnd in the foreground even if the screen turns off has worked well for me for now. I've also done what @schubeda suggested about optimization.

Some of the same sections where I go through, still don't register but I don't know if it's OsmAnd or the satellite quality of that area.

I have yet to test if it logs with the app in the background (i.e. closed).

**Reply by schubeda** [reply]
Thanks for the info. I just tried with OsmAnd+ 4.1.11, still under Android 10. Still it doesn't work. As soon as I switch off the screen during navigation, GPS is lost.

**Comment by gsosna** [comment]
Could be related to missing ACCESS_BACKGROUND_LOCATION permission, similar to https://github.com/osmandapp/OsmAnd/discussions/11816

**Reply by schubeda** [reply]
True, location is only allowed while using the app. But it should still work, because the other app I mentioned doesn't have permission for background location either. A notification seems to be sufficient: https://github.com/osmandapp/OsmAnd/discussions/11816#discussioncomment-964364

Also, I don't see a way to enable background location.

**Reply by Lee-Carre** [reply]
When recording a track, that activity is run as a foreground service. At least that's my understanding of how OsmAnd does things lately, according to the developers.

**Reply by schubeda** [reply]
@Lee-Carre 
Track recording wasn't my original concern. Not sure if this is related.

But you are right, such things should be done with a foreground service. Like this, my navigation should work when the screen is turned off - but it doesn't.

**Comment by Lee-Carre** [comment]
Simply use something like [My Location](/mirfatif/MyLocation) (available via F-Droid), which has a (simple toggle button) feature to keep the GNSS receiver enabled.

I rely on it for (among other things) surveying, and it avoids a whole bunch of problems.

When you're finished travelling, don't forget to allow your GNSS chip to turn off 🙂.

**Reply by schubeda** [reply]
Thanks for the workaround. I will probably try it.

**Reply by Lee-Carre** [reply]
> workaround

I'd argue that it's not a workaround, but the actual solution which you seek.

Let's say that the developers did as you wished, and added the `background location` permission. This still wouldn't reliably solve the problem you describe, because (to my understanding) just because an app *may* request location in the background doesn't mean that it is or will (unless it decides that it needs to).
The intended use-case is for something like a weather app, in order for it to auto-magically fetch weather for the current location, without requiring user-intervention (or waiting until the user opens the app before doing all that updating).
For the likes of a navigation app, what use is fetching location unless the user is actually using the app?

Now, you might argue something like track-recording or such. Well, that's a different function, which could (and arguably should) be done by a separate app. But, that's a different argument. In the original context of navigation, background-fetching of location isn't needed.
An app may also not query an active provider, but use use passive provider (essentially a cache of past location fixes). So, this extra permission you seek is different to what you're imagining, and wanting to do.

Another factor, to consider, as to why background location queries are restricted, is security / privacy, and avoiding abuse and/or general misuse.

So, it seems to me that OsmAnd is doing as it should. Keeping the receiver active does use a non-trivial amount of energy (especially with duty-cycling disabled).

Your case, to me, simply seems to be one of wanting the receiver to stay on, while you're out, so that stabilised position is instantly available to any app. That's exactly what My Location (or similar) will do for you.

This keeps OsmAnd itself simpler, for those who value battery-saving over performance.

For example, one trade-off of configuring for performance is that I must carry around a larger (external) battery pack. But, I want to be limited by factors other than power, to do lots of surveying for hours (and an internal ~3Ah wouldn't last anywhere near long enough, with everything I have running (I collect more than just map data)). So, having accepted the need for greater power reserves, it's trivial for me to then be unconcerned about usage, and enable modes & features which use more energy. For me, that's worth it. But, not for others.

Quite often, in multiple projects, I see complaints about ‘battery drain’. For those users, energy-efficiency is much more of a priority.

What's good for one, isn't necessarily good for others. Adjust your circumstances as needed.
You want the receiver always-on; so keep it always-on. Problem solved (without causing any other problems, for anyone else).

Plus, this technique works for *any* app (or, actually, **all** apps), rather than each of them having to fiddle their code to achieve what is better done by a dedicated app.
Yet another advantage is that even if all your apps would keep the receiver enabled, what happens when you switch between them? Using My Location (or similar) solves *that*, too.
I mentioned, before, about apps querying the passive provider; keeping GNSS enabled also keeps the passive cache updated for any apps which use it; so that e.g. your weather app always knows where you are (rather than where you were when you last checked the map or something which triggered an active query).

Everything just works, without adding needless complexity to every other app.

Solution. Not workaround.

Messing with permissions is the workaround, actually.

**Reply by Lee-Carre** [reply]
Since this solves the problem, please up-vote & mark-as-answer (if available for non-question-type discussions) for the benefit of others.

**Reply by schubeda** [reply]
> Let's say that the developers did as you wished, and added the background location permission.

I never said I wanted to add that permission. On the contrary, I explained to @gsosna that this permission is not needed. HERE WeGo can do it without background location. Osmand used to work without additional app before I installed a new OS on my smartphone. Osmand is clearly intended to be able to do navigation while the screen is turned off.

> Your case, to me, simply seems to be one of wanting the receiver to stay on, while you're out, so that stabilised position is instantly available to any app.

I'm not out. I'm talking about an active navigation. It's a feature of Osmand that you can turn off the screen while navigating in order to save battery. It should work. It's clearly a bug.

**Comment by schubeda** [comment]
It seems like I'm not the only one. Should this be moved back to issues?

**Reply by vshcherb** [reply]
And what it will do in issues then? We keep in the issues only what could be fixed with a known solution or what's affecting at least 10% of all user base (this is definitely not a case)

**Reply by schubeda** [reply]
If these are the official rules, you are right, this is not a case.

**Reply by vshcherb** [reply]
Developers need all comments to issues and discussions, so there is no difference regarding that point
